pkg/template/servicebroker/catalog.go

```diff
@@ -108,10 +108,14 @@ func serviceFromTemplate(template *templateapi.Template) *api.Service {
 }
 
 func (b *Broker) Catalog() *api.Response {
+	if err := b.waitForSync(); err != nil {
+		return api.InternalServerError(err)
+	}
```

Comment:
is this strictly necessary?  the catalog repeatedly polls the broker, so if we miss some entries because the cache was stale, is that actually a big deal?  (assuming i'm understanding the intent here)